### PR TITLE
add script to update submodule 

### PR DESCRIPTION
**Pull request by nopeppermint**
add a script to update submodule c_library in https://github.com/mavlink/c_uart_interface_example.git on each new mavlink build

=> someone could retest please

### CODE REVIEW

**Comment by nopeppermint** [comment]
this solve https://github.com/mavlink/c_uart_interface_example/issues/14 I think
and replace https://github.com/mavlink/c_uart_interface_example/pull/13 (Readme should be updated again anyway, if the submodule of c_uart_interface_example is generated automatically)

File: scripts/update_submodule_c_uart_interface_example.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# to repository whenever the mavlink message specifications are updated.
+
+# settings
+C_UART_INTERFACE_EXAMPLE_GIT_REMOTENAME=origin
+C_UART_INTERFACE_EXAMPLE_GIT_BRANCHNAME=master
+C_UART_INTERFACE_EXAMPLE_PATH=$PWD/c_uart_interface_example
+SUBMODULE_PATH=mavlink/include/mavlink/v1.0
+
+# clone 
+git clone https://github.com/mavlink/c_uart_interface_example.git
```
Comment (dagar) [inline scripts/update_submodule_c_uart_interface_example.sh]: Why not just git clone --recursive?

Comment (nopeppermint) [inline scripts/update_submodule_c_uart_interface_example.sh]: I didn't know this command, but this work too.

**Comment by aerialhedgehog** [comment]
i'm not too familiar with this script, but it seems in this case it's probably simplest to leave it as part of the readme file for the user to do the submodule update manually

**Comment by nopeppermint** [comment]
@aerialhedgehog what speaks again a c_uart_interface_example which works out of the box whit the current mavlink version?

**Comment by LorenzMeier** [comment]
What is the testing level of this?

**Comment by nopeppermint** [comment]
it's tested under a local linux, but as I have no access to the c_uart_interface_example the push failed
Would be nice if someone else could retest this.
```
 $ ./update_submodule_c_uart_interface_example.sh 
Nach »c_uart_interface_example« wird geklont
remote: Counting objects: 1136, done.
remote: Total 1136 (delta 0), reused 0 (delta 0), pack-reused 1136
Objekte werden empfangen: 100% (1136/1136), 971.31 KiB | 590.00 KiB/s, done.
Unterschiede werden aufgelöst: 100% (742/742), done.
Verbundenheit wird überprüft … Fertig.
Submodul 'mavlink/include/mavlink/v1.0' (git://github.com/mavlink/c_library.git) für Pfad 'mavlink/include/mavlink/v1.0' in die Konfiguration eingetragen.
Nach »mavlink/include/mavlink/v1.0« wird geklont
remote: Counting objects: 4310, done.
remote: Compressing objects: 100% (124/124), done.
remote: Total 4310 (delta 101), reused 0 (delta 0), pack-reused 4186
Objekte werden empfangen: 100% (4310/4310), 1.73 MiB | 1.32 MiB/s, done.
Unterschiede werden aufgelöst: 100% (3832/3832), done.
Verbundenheit wird überprüft … Fertig.
Submodul-Pfad: 'mavlink/include/mavlink/v1.0': '7d0aa0205ef8bd13bf855b21d3ebe054dc9109ef' ausgecheckt
Betrete 'mavlink/include/mavlink/v1.0'
Von git://github.com/mavlink/c_library
 * branch            master     -> FETCH_HEAD
Aktualisiere 7d0aa02..2761697
Fast-forward
 ASLUAV/ASLUAV.h                                    |   84 +-
 ASLUAV/mavlink_msg_asl_obctrl.h                    |  377 +++++++
 ASLUAV/mavlink_msg_ekf_ext.h                       |  353 +++++++
 ASLUAV/mavlink_msg_sens_atmos.h                    |  233 +++++
 ASLUAV/mavlink_msg_sens_batmon.h                   |  497 +++++++++
 ASLUAV/testsuite.h                                 |  202 ++++
 ASLUAV/version.h                                   |    2 +-
 ardupilotmega/ardupilotmega.h                      |  180 +++-
 ardupilotmega/mavlink_msg_ahrs3.h                  |  425 ++++++++
 .../mavlink_msg_autopilot_version_request.h        |  233 +++++
 ardupilotmega/mavlink_msg_ekf_status_report.h      |  329 ++++++
 .../mavlink_msg_gimbal_axis_calibration_progress.h |  257 +++++
 ardupilotmega/mavlink_msg_gimbal_control.h         |  305 ++++++
 .../mavlink_msg_gimbal_erase_firmware_and_config.h |  257 +++++
 .../mavlink_msg_gimbal_factory_parameters_loaded.h |  209 ++++
 ...ink_msg_gimbal_home_offset_calibration_result.h |  209 ++++
 .../mavlink_msg_gimbal_perform_factory_tests.h     |  233 +++++
 ardupilotmega/mavlink_msg_gimbal_report.h          |  473 +++++++++
 ...link_msg_gimbal_report_factory_tests_progress.h |  281 ++++++
 ardupilotmega/mavlink_msg_gimbal_reset.h           |  233 +++++
 .../mavlink_msg_gimbal_set_factory_parameters.h    |  521 ++++++++++
 .../mavlink_msg_gimbal_set_home_offsets.h          |  233 +++++
 ardupilotmega/mavlink_msg_gopro_command.h          |  305 ++++++
 ardupilotmega/mavlink_msg_gopro_power_off.h        |  233 +++++
 ardupilotmega/mavlink_msg_gopro_power_on.h         |  233 +++++
 ardupilotmega/mavlink_msg_gopro_response.h         |  305 ++++++
 ardupilotmega/mavlink_msg_led_control.h            |  321 ++++++
 ardupilotmega/mavlink_msg_mag_cal_progress.h       |  393 ++++++++
 ardupilotmega/mavlink_msg_mag_cal_report.h         |  521 ++++++++++
 ardupilotmega/mavlink_msg_mount_status.h           |   30 +-
 ardupilotmega/mavlink_msg_pid_tuning.h             |  353 +++++++
 ardupilotmega/testsuite.h                          | 1063 ++++++++++++++++++++
 ardupilotmega/version.h                            |    2 +-
 autoquad/autoquad.h                                |   46 +-
 autoquad/mavlink_msg_aq_esc_telemetry.h            |  397 ++++++++
 autoquad/testsuite.h                               |   52 +
 autoquad/version.h                                 |    2 +-
 checksum.h                                         |    6 +
 common/common.h                                    |  104 +-
 common/mavlink_msg_actuator_control_target.h       |  249 +++++
 common/mavlink_msg_att_pos_mocap.h                 |  297 ++++++
 common/mavlink_msg_autopilot_version.h             |  288 +++++-
 common/mavlink_msg_camera_trigger.h                |  233 +++++
 common/mavlink_msg_distance_sensor.h               |   10 +-
 common/mavlink_msg_global_position_int.h           |   10 +-
 common/mavlink_msg_gps2_raw.h                      |   10 +-
 common/mavlink_msg_gps_global_origin.h             |   10 +-
 common/mavlink_msg_gps_raw_int.h                   |   10 +-
 common/mavlink_msg_hil_gps.h                       |   10 +-
 common/mavlink_msg_landing_target.h                |  305 ++++++
 common/mavlink_msg_local_position_ned_cov.h        |  178 +++-
 common/mavlink_msg_param_map_rc.h                  |  393 ++++++++
 common/mavlink_msg_position_target_global_int.h    |   10 +-
 common/mavlink_msg_radio_status.h                  |   70 +-
 common/mavlink_msg_scaled_imu3.h                   |  425 ++++++++
 common/mavlink_msg_scaled_pressure2.h              |  281 ++++++
 common/mavlink_msg_set_actuator_control_target.h   |  297 ++++++
 common/mavlink_msg_set_gps_global_origin.h         |   10 +-
 .../mavlink_msg_set_position_target_global_int.h   |   10 +-
 common/mavlink_msg_vibration.h                     |  353 +++++++
 common/testsuite.h                                 |  470 ++++++++-
 common/version.h                                   |    2 +-
 matrixpilot/matrixpilot.h                          |   22 +-
 matrixpilot/version.h                              |    2 +-
 mavlink_conversions.h                              |   78 +-
 mavlink_helpers.h                                  |  229 +++--
 mavlink_types.h                                    |   43 +-
 minimal/version.h                                  |    2 +-
 pixhawk/pixhawk.h                                  |   22 +-
 pixhawk/version.h                                  |    2 +-
 protocol.h                                         |    6 +
 slugs/slugs.h                                      |   22 +-
 slugs/version.h                                    |    2 +-
 test/version.h                                     |    2 +-
 ualberta/ualberta.h                                |    6 +-
 ualberta/version.h                                 |    2 +-
 76 files changed, 14481 insertions(+), 382 deletions(-)
 create mode 100644 ASLUAV/mavlink_msg_asl_obctrl.h
 create mode 100644 ASLUAV/mavlink_msg_ekf_ext.h
 create mode 100644 ASLUAV/mavlink_msg_sens_atmos.h
 create mode 100644 ASLUAV/mavlink_msg_sens_batmon.h
 create mode 100644 ardupilotmega/mavlink_msg_ahrs3.h
 create mode 100644 ardupilotmega/mavlink_msg_autopilot_version_request.h
 create mode 100644 ardupilotmega/mavlink_msg_ekf_status_report.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_axis_calibration_progress.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_control.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_erase_firmware_and_config.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_factory_parameters_loaded.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_home_offset_calibration_result.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_perform_factory_tests.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_report.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_report_factory_tests_progress.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_reset.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_set_factory_parameters.h
 create mode 100644 ardupilotmega/mavlink_msg_gimbal_set_home_offsets.h
 create mode 100644 ardupilotmega/mavlink_msg_gopro_command.h
 create mode 100644 ardupilotmega/mavlink_msg_gopro_power_off.h
 create mode 100644 ardupilotmega/mavlink_msg_gopro_power_on.h
 create mode 100644 ardupilotmega/mavlink_msg_gopro_response.h
 create mode 100644 ardupilotmega/mavlink_msg_led_control.h
 create mode 100644 ardupilotmega/mavlink_msg_mag_cal_progress.h
 create mode 100644 ardupilotmega/mavlink_msg_mag_cal_report.h
 create mode 100644 ardupilotmega/mavlink_msg_pid_tuning.h
 create mode 100644 autoquad/mavlink_msg_aq_esc_telemetry.h
 create mode 100644 common/mavlink_msg_actuator_control_target.h
 create mode 100644 common/mavlink_msg_att_pos_mocap.h
 create mode 100644 common/mavlink_msg_camera_trigger.h
 create mode 100644 common/mavlink_msg_landing_target.h
 create mode 100644 common/mavlink_msg_param_map_rc.h
 create mode 100644 common/mavlink_msg_scaled_imu3.h
 create mode 100644 common/mavlink_msg_scaled_pressure2.h
 create mode 100644 common/mavlink_msg_set_actuator_control_target.h
 create mode 100644 common/mavlink_msg_vibration.h
[master dcfaca8] Mavlink c_library Submodule updated to rev https://github.com/mavlink/c_library/tree/27616975e3a6d38937134784bed3393f26f7477c
 1 file changed, 1 insertion(+), 1 deletion(-)
Username for 'https://github.com': nopeppermint
Password for 'https://nopeppermint@github.com': 
remote: Permission to mavlink/c_uart_interface_example.git denied to nopeppermint.
fatal: unable to access 'https://github.com/mavlink/c_uart_interface_example.git/': The requested URL returned error: 403
```
gitk in the git repo gives me:

```
Autor: nopeppermint <stefan-weber.k@gmx.ch>  2015-06-23 14:49:50
Eintragender: nopeppermint <stefan-weber.k@gmx.ch>  2015-06-23 14:49:50
Eltern: 7aa7dc432b9cb9056bc05d5b837f319b63fd75de (Merge pull request #9 from aerialhedgehog/object)
Zweig: master
Folgt auf: 
Vorgänger von: 

    Mavlink c_library Submodule updated to rev https://github.com/mavlink/c_library/tree/27616975e3a6d38937134784bed3393f26f7477c

------------------------- mavlink/include/mavlink/v1.0 -------------------------
Submodule mavlink/include/mavlink/v1.0 7d0aa02..2761697:
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/047b6fbb16ded3c4e609c4da901691882025403a
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/59ddd8a3d9e2a60bf4e28ea33620cc7424a64d30
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/e6f785ad31344942c10e7541052a7889a87a0b77
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f1fd4cba5bfd72d35f401b0216b3f38d9caedc5b
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/6634482cb997e409640444a86ecdd57bedc22e0b
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/678b2a7e1b08d3e71a53f0173b015fdd05260a18
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/4d9f76451c90cb34d8e05b14a6fa87ae9edc2bf6
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/5af3baba9fdccf0961fc4c103f6627ba099cc655
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/fda33552fa6db8ac6f066959c7d30c4b27a0c0df
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/1d9da406351d061167891f0ed908177655d34079
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/d546467f301b90186fb13ece9b8915a1703c3ab0
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/daae22e5425349c275fcf5505e3d5e126d8d44f1
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/5c3b98074f25b08d3e053faf2717335879f40a05
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/3123dac07517256117c81ec36df3e686f978ebe0
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f6e4de5be43056a8309fad7d9fbe0ad8c06c6c61
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/d707c6d94ab7067b4c8a946784c01c9f2a3e1f0b
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f76b6acc844960d7dbf3a726daef90749de223b9
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/5030068fd1cef668186bb2a8868cac26f6e70634
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/8f06164acb13a5abae1dbb4ed73ece622ebec163
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/9169a1f0b8d43240c61af460effa7bac513a2dd1
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/2f18d5410b0014311383297994f06b42f0cd26ab
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/df1bcab88facfc0f319c311582b3f636ec9c6b48
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/c2c22d643478cf1505137ce90557e3e803721979
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/919a57754db06d79df9983beabd4e67a21f2ed84
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/71632b0ec376ed6328abeffcc7ba3578754042e8
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/9b885358bbbe94f049a3303fa4a7829061b6fe4a
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/e485a67e4e9c1dd2e96acc67a229daf49893aaa1
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/e1048ab179419015b2c51b59f020ca052371dec9
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/e42d4d1cf7956a5245f3bf40dbdbea6decc0109d
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/c6b5a128479bd36c8389952172dc8efac8069aed
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/81c642c9907da3b9073192a96e1bffbe3fea9bb1
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/97903b944076c2b967997812d4131b1794d47dbd
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/19d965c1727856ad5846e97389ae26cede9e5ccc
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/3cc380c666feb267f36e05af25baa312e2352eca
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/6365256d672ec81cd68f3d1913d95d43375f1678
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/54e826fb8037dfa8dcdd2ce0de6d9bf5dd783048
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/ed2f3e442097de1745b51c02a621b7018990a6ff
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/7a7ce3362706dd0164377d6182ce73f22f4d41ef
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/b0a85a7589738bd2ce3d63bebb8d36eafcef70ce
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/ce2d5d83b62af42e314b8911f298daf55c19f9fe
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/45cd8e802a1c7f9b96b37a06faa5dbd7901e9f2c
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/d89a2c10199462765310ae679a1b8498c647055b
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/b160f0e3c7174c115117eece93be691488faaf51
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/447b2e07daac0756d2aeb4841fbb6ba3d7741225
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/b92a26fcf487e737b5229fbeb8d19168145ab969
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/88edd139c6c224aa39354a5d8f4d2dd401f9ea4e
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/a581f0bab1d1a8ecceeea3c8a3abbcc0d3904dc2
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/814b9654e91b806e759c9b888e9420c81024cf13
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/5470db218334756e9c4c7e64f4462535583e170f
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/de22c72f79b34e71f7c253f3d1988e5ea6dce223
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/6e51562cfb98edeca4ee70717f7f4c56b9adc728
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f8c2c09db537c9940ef8e4a58b8866bca1df65eb
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/75f4879ac745398d62583c88a380cd41ae5979f7
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f707c32fc645510e887b6967663149126b827ff8
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/dc3c4a96514e5a5ce0a8103ca16c4668d5804a68
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/bcbc44997330b4ebfaa40d23d7391008cbaf572e
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/d93446c20cacb5b5e94763667a4317d0b4a4ac5e
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/733703f34d59226a61c04ce1cda7313ebdb91353
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/b6ad131b38b958d19cc825a81b50ead5148564ed
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/4228349fecbb46c719b236d38c1beb8da87ea80e
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/837eb2db4dcc65904c5d4294ed3e64001128aa4f
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/87f912d4e7eaa0f46d2e5a20997356cff6b55a39
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/c1e6045c467c6774a6e8a786a29c0eafaeeb0b14
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/1014f7e28b1b16915c694307b72178b52c2008d1
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/b32a42b199633f7a1b53953a74f643ff22c5ceaf
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/18c7c67905d3e5715edb5eec1b428e30ee570c38
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/7a6380100d03da571fd185a34198a6a6244e8f9f
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/91fa66739ae45876b924908afe8d62f7e9d489d5
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/983f4b58aeeeffbd833add74aaff606ad37384af
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/76a2d91b405c9333caf1497a32a6d8109620acf5
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/f057d511fe554d50e1dbd0ac5530b81026f1fbd5
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/e38c089f98b441e6b3925b71a5be50908949ce56
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/4f44e4797bc16a05b101d6efe7555df4bf24a3ef
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/32379d5b25399d4884335fcf79f12078cd9709e9
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/43a7fd8e23773c0879ddfffddc5f8aded0eb6d4a
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/22ab243c8a1b24596ecc2714869333502331b869
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/bb81de90a535555218caf0e47ec359c0d5cff29c
  > autogenerated headers for rev https://github.com/mavlink/mavlink/tree/83b3c23332bab1b320b9674e604d90b3fc482f75
```

**Pull request closed without merging**